### PR TITLE
[CN-672] Add page with Native Memory parameters

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**

### CODE REVIEW

**Review by cagric0 — APPROVED** [review]
LGTM, but I think you should also get a review from Angela.

**Review by hasancelik — APPROVED** [review]
LGTM 👍 as @cagric0 said, getting reviews from doc team would be good before the merge.

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:2]:
```suggestion
:description: The Native Memory Store is a pluggable memory manager. It allows you to set up different configurations, or stores, for your application’s data structures on a single JVM. These memory stores form part of a common access layer which scales up to use a configured amount of memory on the JVM.
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:6]:
```suggestion
As well as providing more predictable application scaling, using a Native Memory Store boosts performance, reduces latency, and minimizes garbage collection pauses. 
```
If we change the title, I don't think the original sentence is needed.

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:1]:
```suggestion
= Configuring the Native Memory Store
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:8]:
```suggestion
== Configuring a Native Memory Store
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:10]:
```suggestion
Below are the configuration options for the Native Memory Store.
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:17]:
```suggestion
| Specifies the mechanism for allocating memory; either XXXX or XXXX.
```
It makes sense to give both options if they are available, and to explain what they are used for.

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:23]:
```suggestion
|Size of smallest block of memory to allocate.
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:26]:
```suggestion
|Size of the page in bytes to allocate as a block of memory.
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:29]:
```suggestion
|Percentage of the allocated native memory that is used for the metadata of other map components such as index (for predicates), offset, etc.
```
What if the data structure isn't a map?

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:37]:
```suggestion
- Creates a cluster with `Native Memory` and the `POOLED` allocator enabled.
```

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:38]:
```suggestion
- Configures a map on the cluster to use native memory.
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
I'm not sure about the introduction/description as I read had to read it a few times before I understood the advantages of a native memory store. I tried to reword but not sure if it is accurate.

File: docs/modules/ROOT/pages/native-memory.adoc
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:17]: For this line, you need to add in the name of the two allocator types. It looks like the first is `POOLED` but I'm not sure about the second one.

**Review by angelasimms — CHANGES_REQUESTED** [review]

**Pull request merged**
